### DISCUSSION TITLE
Refresh relationmanager table from page header action

### DISCUSSION THREAD

**Discussion by jwhoami** [filamentphp/filament#14894] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

Hi,

I want to click on a page action (shown on top of the page) and dispatch an event which refreshes the relationmanager on the page below.

How can i achieve this?

Thanks

**Comment by leandrocfe** [comment, marked as answer]
Add it to the Relation Manager Class

```php
use Livewire\Attributes\On;

#[On('refreshData')]
public function refreshData(): void
{
    // do something
}
```

Dispatch it in the Action
```php
Actions\Action::make('yourAction')
    ->dispatch('refreshData')
```

**Reply by jwhoami** [reply]
Thanks leandrocfe.

**Reply by jwhoami** [reply]
Just to know if it's possible...

I was doing $this->dispatch('$refresh') on the page action which does refresh the page. 
However the relationmanager was not getting refreshed and so I asked the original question.

Would it possible to make the relationmanager to react $refresh event? That way I don't have to
dispatch 2 events.

Thanks.
